From 1242f9f7018c6bf10d9cb03777f58ecbb5ea1235 Mon Sep 17 00:00:00 2001
From: Paulo Ewerton <pauloewerton@lsd.ufcg.edu.br>
Date: Mon, 9 Nov 2015 14:03:45 +0000
Subject: [PATCH] Retrieve AUTH_INTERFACE object from keystoneauth

This patch changes the AUTH_INTERFACE object to
keystoneauth1.plugin.AUTH_INTERFACE so that it passes the
check at keystoneauth1.identity.base.get_endpoint.

Co-Authored-By: Doug Fish <drfish@us.ibm.com>

Change-Id: Iaf48707ef82c1d4b873e07644580dd9572ce0992
Partial-Bug: #1471943
---
 keystoneclient/auth/base.py | 3 ++-
 requirements.txt            | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/keystoneclient/auth/base.py b/keystoneclient/auth/base.py
index 27faa2a..369fc55 100644
--- a/keystoneclient/auth/base.py
+++ b/keystoneclient/auth/base.py
@@ -15,13 +15,14 @@ import os
 import six
 import stevedore
 
+from keystoneauth1 import plugin
 from keystoneclient import exceptions
 
 
 # NOTE(jamielennox): The AUTH_INTERFACE is a special value that can be
 # requested from get_endpoint. If a plugin receives this as the value of
 # 'interface' it should return the initial URL that was passed to the plugin.
-AUTH_INTERFACE = object()
+AUTH_INTERFACE = plugin.AUTH_INTERFACE
 
 PLUGIN_NAMESPACE = 'keystoneclient.auth.plugin'
 IDENTITY_AUTH_HEADER_NAME = 'X-Auth-Token'
diff --git a/requirements.txt b/requirements.txt
index 808053d..9f19ff9 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -8,6 +8,7 @@ argparse
 Babel>=1.3
 iso8601>=0.1.9
 debtcollector>=0.3.0 # Apache-2.0
+keystoneauth1>=1.0.0
 netaddr!=0.7.16,>=0.7.12
 oslo.config>=2.3.0 # Apache-2.0
 oslo.i18n>=1.5.0 # Apache-2.0
-- 
1.9.1

